modules/porous_flow/src/actions/PorousFlowAddBCAction.C

```diff
@@ -0,0 +1,102 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PorousFlowAddBCAction.h"
+#include "PorousFlowDictator.h"
+#include "NonlinearSystemBase.h"
+
+registerMooseAction("PorousFlowApp", PorousFlowAddBCAction, "add_porous_flow_bc");
+
+InputParameters
+PorousFlowAddBCAction::validParams()
+{
+  InputParameters params = MooseObjectAction::validParams();
+  params.addClassDescription("Action that allows adding BCs using proxy classes.");
+  return params;
+}
+
+PorousFlowAddBCAction::PorousFlowAddBCAction(const InputParameters & parameters)
+  : MooseObjectAction(parameters)
+{
+}
+
+void
+PorousFlowAddBCAction::act()
+{
+  if (_type == "PorousFlowSinkBC")
+    setupPorousFlowEnthalpySink();
+  // NOTE: no reason to catch unknown types here - MOOSE will catch that much earlier
+}
+
+void
+PorousFlowAddBCAction::setupPorousFlowEnthalpySink()
+{
+  std::string nm = name();
+
+  THREAD_ID tid = 0;
+  const InputParameters & obj_pars = getObjectParams();
+
+  // check that we have 2 porous flow variables
+  const PorousFlowDictator & dictator = _problem->getUserObject<PorousFlowDictator>(
+      obj_pars.get<UserObjectName>("PorousFlowDictator"));
+  if (dictator.numVariables() != 2)
+    mooseError(name(),
+               ": Number of porous flow variables in simulation is '",
+               dictator.numVariables(),
+               "', but this BC can be used only with pressure and temperature.");
```

Comment:
If you like, you could add something like "The Dictator will take care of you" to this error message.

Reply:
I would like that, but people were lately sensitive to this kind of humor, so I am afraid we will have to let this one slide :-(

Reply:
Interesting